src/components/ShowOnClick.js

```diff
@@ -0,0 +1,63 @@
+import mixinDeep from 'mixin-deep'
+import { Component } from 'react'
+import { div, hh } from 'react-hyperscript-helpers/lib/index'
+
+
+/**
+ * @param [containerProps]
+ * @param button
+ * @param [bgProps]
+ * @param [closeOnClick=true]
+ * @param [closeOnEsc=true]
+ * @param children
+ */
+export default hh(class ShowOnClick extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      hidden: true
+    }
+  }
+
+  setVisibility = (visible) => {
+    this.setState({ hidden: !visible })
+  }
+
+  listenForEscape = (e) => {
+    if (e.key === 'Escape') {
+      window.removeEventListener('keydown', this.listenForEscape)
```

Comment:
What happens if this component gets unmounted while open? Will the event listener still get removed?

Reply:
Added `willUnmount` to handle this